services/toDoService.js

```diff
@@ -14,3 +32,59 @@ export async function getAllToDos() {
 
   return sanitizedToDos;
 }
+
+export async function getToDosByDestinationId(destinationId) {
+  if (!validateId(destinationId)) return { status: 400, error: "Invalid destinationId" };
+  await dbConnect();
+
+  const toDos = await ToDo.find({ destinationId: destinationId });
+  if (!Array.isArray(toDos)) return { status: 500, error: "Internal server error" };
+  const sanitizedToDos = toDos.map((toDo) => ({
+    id: toDo.id,
+    description: toDo.description,
+    checked: toDo.checked,
+    destinationId: toDo.destinationId,
+  }));
+
+  return sanitizedToDos;
+}
+
+export async function createToDo(body, destinationId) {
+  if (!validateId(destinationId)) return { status: 400, error: "Invalid destinationId" };
+  await dbConnect();
+
+  const newToDo = await ToDo.create({
+    description: body.description,
+    destinationId: destinationId,
+  });
+
+  return await getToDosByDestinationId(destinationId);
+}
+
+export async function updateToDo(id, body) {
+  if (!validateId(id)) return { status: 400, error: "Invalid id" };
+
+  await dbConnect();
+  const toDo = await ToDo.findById(id);
+  if (!toDo) return { status: 404, error: id + " not found" };
+  if (body.description) toDo.description = body.description;
+  toDo.checked = body.checked;
+  const savedToDo = await toDo.save();
+  if (!savedToDo) return { status: 500, error: "Internal server error" };
+  return await getToDosByDestinationId(toDo.destinationId);
+}
+
+export async function deleteToDo(id, destinationId) {
+  if (!validateId(id)) return { status: 400, error: "Invalid id" };
+  if (!validateId(destinationId)) return { status: 400, error: "Invalid destinationId" };
```

Comment:
If a todo has an invalid destinationId should be checked elsewhere (when creating the todo). This is the wrong place. If a Todo in fact has an invalid destinationId you would also be unable to delete it here because of this check.

Reply:
Implemented validity db-check here as well.